core/lib/dal/src/via_blocks_web3_dal.rs

```diff
@@ -0,0 +1,136 @@
+use zksync_db_connection::{connection::Connection, error::DalResult, instrument::InstrumentExt};
+use zksync_types::{
+    api::{BlockDetails, L1BatchDetails},
+    L1BatchNumber, L2BlockNumber,
+};
+
+pub use crate::models::storage_block::{L1BatchMetadataError, L1BatchWithOptionalMetadata};
+use crate::{
+    models::storage_btc_block::{ViaStorageBlockDetails, ViaStorageL1BatchDetails},
+    Core,
+};
+
+#[derive(Debug)]
+pub struct ViaBlocksWeb3Dal<'a, 'c> {
+    pub(crate) storage: &'a mut Connection<'c, Core>,
+}
+
+impl ViaBlocksWeb3Dal<'_, '_> {
+    pub async fn get_l1_batch_details(
+        &mut self,
+        l1_batch_number: L1BatchNumber,
+    ) -> DalResult<Option<L1BatchDetails>> {
+        let l1_batch_details = sqlx::query_as!(
+            ViaStorageL1BatchDetails,
+            r#"
+            WITH
+                mb AS (
+                    SELECT
+                        l1_gas_price,
+                        l2_fair_gas_price,
+                        fair_pubdata_price
+                    FROM
+                        miniblocks
+                    WHERE
+                        l1_batch_number = $1
+                    LIMIT
+                        1
+                )
+            SELECT
+                l1_batches.number,
+                l1_batches.timestamp,
+                l1_batches.l1_tx_count,
+                l1_batches.l2_tx_count,
+                l1_batches.hash AS "root_hash?",
+                commit_history.reveal_tx_id AS "commit_tx_hash?",
+                commit_history.confirmed_at AS "committed_at?",
+                proof_history.reveal_tx_id AS "prove_tx_hash?",
+                proof_history.confirmed_at AS "proven_at?",
+                bir.is_finalized AS "is_finalized?",
```

Comment:
It looks that we don't update this field properly and the result is that the batch always has a status `sealed`